falak/pkg/geolocation/client/client.go

```diff
@@ -0,0 +1,44 @@
+package client
+
+import (
+	"context"
+	"encoding/json"
+	"fmt" // Import your custom client package
+	"io"
+
+	"github.com/jadwalapp/symmetrical-spoon/falak/pkg/httpclient"
+)
+
+type client struct {
+	cli httpclient.HTTPClient
+}
+
+// GetGeoLocation fetches the location data using your custom HTTP client
+func (c *client) GetGeoLocationInfo(ctx context.Context, r *GetGeoLocationInfoRequest) (*GetGeoLocationInfoResponse, error) {
+	// Send the request using the custom client
+	resp, err := c.cli.Get(fmt.Sprintf("https://freeipapi.com/api/json/%s", r.Ip), nil, nil)
```

Comment:
please put this similar to prayer time one, in `falak.env` and read it from there to be able to change it when testing.

also pass it from `client` struct under `baseUrl`.

look at google client as an example.